New section for Router Protocol (#974)
oasisprotocol/docs: docs/dapp/opl/router-protocol/README.mdx

2. Orchestrators listen to the event emitted by the Gateway contract.
3. Relayer picks up the transaction signed by the orchestrator and forwards the message to the destination chain's Router Gateway contract.
4. Gateway contract on the destination chain calls dApps contract `iReceive` function.
5. For the acknowledge message, the flow is reversed and the relayer calls on the source chain the `iAck` function on the dApps contract.

### Fees

Two actors are paying the fees during the cross-chain messaging.
- The dApp **user** pays when initiating the transaction on the source chain
- The dApp **fee payer** pre-pays the relayers for calling the Router Gateway contract

That the correct fee payer is used, the dApp's contract has to register the fee payer address as metadata to the Router Gateway.
Additionally the fee payer need to be approved on the Router, which can be done on the [Router Explorer].

For more info about the [fee management] conduct the Router documentation.

[Router Explorer]: https://testnet.routerscan.io/feePayer
[fee management]: https://docs.routerprotocol.com/develop/message-transfer-via-crosstalk/key-concepts/fee-management

:::note Example: PingPong

> Check out our [PingPong example]

Reviewer: Link is broken

---
description: Cross-chain dapps with Router Protocol
---
import DocCard from '@theme/DocCard';
import DocCardList from '@theme/DocCardList';
import {findSidebarItem} from '@site/src/sidebarUtils';

# Router Protocol

Router Protocol has two different frameworks:

> - **Router CrossTalk**: A stateless and statefull cross-chain messaging

Reviewer 2: ```suggestion
- **Router CrossTalk**: A stateless and stateful cross-chain messaging
```

- **Router Nitro**: A native cross-chain asset-transfer bridge

Router wrote a [guide] to help you choosing the right framework.

In our documentation we will focus on Router CrossTalk.

> If you are more interested in asset transfers, please refer to the [Router Nitro documentation] by Router Protocol

Reviewer 2: Please wrap the whole file to 80 characters or so for easier review.

[Router Nitro documentation]: https://docs.routerprotocol.com/develop/category/asset-transfer-via-nitro
[guide]: https://docs.routerprotocol.com/overview/choosing-the-right-framework

## Router CrossTalk

The CrossTalk framework is designed to facilitate cross-chain interactions, 
enabling developers to create decentralized applications (dApps) 
that can operate across multiple blockchain networks. 
This framework supports both stateless and stateful operations, allowing for flexible 
and efficient communication between contracts on different chains.

### Architecture

![Router Architecture](../../images/opl/router-architecture.png)

The CrossTalk infrustructure consists mainly of three actors
- **Gateway** contracts on source and destination chain
- **Orchestrators** on the Router chain
- **Relayers** who forward the messages to the Router Gateway contracts

The high level flow looks as follows:
1. dApp contract calls `iSend` function on source chain Gateway contract.

> 2. Orchestrators listen to the event emited by the Gateway contract.

Reviewer 2: ```suggestion
2. Orchestrators listen to the event emitted by the Gateway contract.
```

> 3. Relayer pick up the transaction signed by the orchestrator and forward the message to the destination chain's Router Gateway contract.

Reviewer 2: ```suggestion
3. Relayer picks up the transaction signed by the orchestrator and forwards the message to the destination chain's Router Gateway contract.
```

4. Gateway contract on the destination chain calls dApps contract `iReceive` function.
5. For the acknowledge message, the flow is reversed and the relayer calls on the source chain the `iAck` function on the dApps contract.

### Fees

Two actors are paying the fees during the cross-chain messaging.
- The dApp **user** pays when initiating the transaction on the source chain
- The dApp **fee payer** pre-pays the relayers for calling the Router Gateway contract

> That the correct fee payer is used, the dApp's contract has to register the fee payer addresse as metadata to the Router Gateway.

Reviewer 2: ```suggestion
That the correct fee payer is used, the dApp's contract has to register the fee payer address as metadata to the Router Gateway.
```

Additionally the fee payer need to be approved on the Router, which can be done on the [Router Explorer].

For more info about the [fee management] conduct the Router documentation.

[Router Explorer]: https://testnet.routerscan.io/feePayer
[fee management]: https://docs.routerprotocol.com/develop/message-transfer-via-crosstalk/key-concepts/fee-management

## Oasis Example

{/* check out our [PingPong example] */}

[PingPong example]: ./pingpong/README.md

<DocCard item={
    findSidebarItem('/dapp/opl/router-protocol/pingpong/')

> } />

Reviewer 2: Please use the standardized admonition for examples (e.g. https://github.com/oasisprotocol/sapphire-paratime/blob/main/docs/quickstart.mdx?plain=1#L170-L177):

```suggestion
:::note Example: PingPong

Check out our [PingPong example]

:::
```

In the future, we'll introduce a new `:::example` admonition for this purpose.

### Router Protocol Examples

> Further examples you can find in the [Router Protocol documentation]

Reviewer 2: ```suggestion
You can find more examples in the [Router Protocol documentation]
```
